src/ioctls/device.rs

```diff
@@ -207,5 +313,24 @@ mod tests {
 
         assert!(device_fd.has_device_attr(&dist_attr).is_ok());
         assert!(device_fd.set_device_attr(&dist_attr).is_ok());
+
+        // Test `get_device_attr`. Here we try to extract the maximum supported number of IRQs.
+        // This value should be saved in the address provided to the ioctl.
+        let mut data: u32 = 0;
+
+        let mut gic_attr_get = kvm_bindings::kvm_device_attr::default();
+        gic_attr_get.group = KVM_DEV_ARM_VGIC_GRP_NR_IRQS;
+        gic_attr_get.addr = data as u64;
+
+        // Without properly providing the address to where the
+        // value will be stored, the ioctl fails with EFAULT.
+        let res = device_fd.get_device_attr(&mut gic_attr_get);
+        assert!(res.is_err() && errno::Error::last().errno() == 14);
```

Comment:
This should assert on `res`, not on `errno::Error::last()`.

More of a nit, but can you use `libc::EFAULT` instead of magic numbers (i.e. 14)?

Reply:
I think what @dianpopa wants to express here is the ioctl will fail AND the error number in this case (i.e. without properly providing the address to where the value will be stored) should be fixed to 14 (`EFAULT`). Hence we assert both of them by using `&&` here.

I will change the error number :)